snippets/stringToArrayOfWords.md

```diff
@@ -0,0 +1,15 @@
+### stringToArrayOfWords
+
+Converts a given string into an array of words
+
+First create an array of dirt that you want to remove from the string. Then replace each dirt with an empty string. Use `String.split(/[\W_]+/)` to convert the string to an array with the use of regex.
+
+```js
+const stringToArrayOfWords = str => {
+	[".", ",", "?", "!", "& ", "(", ")", "[", "]"].map(d => str = str.split(d).join(""));
```

Comment:
This can be done as a reduce
```js
const stringToArrayOfWords = str => [".", ",", "?", "!", "& ", "(", ")", "[", "]"].reduce( ( s, d ) => s.split( d ).join(""), str).split(/[\W_]+/)
```

Reply:
Also why not just combine the array pattern into a regex?

Reply:
Regex is a pain in some parts of the body :P

Reply:
Writing all the code in one line might not be clear to most of the beginners in javaScript (including me). Certainly cannot be understood in 30 sec :P

Reply:
I made a comment that fixes @atomiks approach for a much easier to understand and faster one liner than the loop version.

By comparison on performance, the loop version is orders of magnitude slower that the non alphabet regex of `/[\W_]/`

Reply:
Working well! and easy to understand too! :)